File: API/Client/StudyList.js
goog.provide('API.Client.StudyList');

/**
 * @record
 */
API.Client.StudyList = function() {}

/**
 * @type {!API.Client.Metadata}
 * @export
 */
API.Client.StudyList.prototype.metadata;

/**
 * @type {!API.Client.Pagination}
 * @export
 */
API.Client.StudyList.prototype.pagination;

/**
 * @type {!Array<!API.Client.Study>}
 * @export
 */
API.Client.StudyList.prototype.data;

File: src/main/scala/org/openapitools/client/api/StudyList.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import StudyList._

case class StudyList (
  metadata: Option[Metadata],
pagination: Option[Pagination],
data: Option[List[Study]])

object StudyList {
  import DateTimeCodecs._

  implicit val StudyListCodecJson: CodecJson[StudyList] = CodecJson.derive[StudyList]
  implicit val StudyListDecoder: EntityDecoder[StudyList] = jsonOf[StudyList]
  implicit val StudyListEncoder: EntityEncoder[StudyList] = jsonEncoderOf[StudyList]
}
